HOMEWORK_30/FinalProject/Game.cpp

```diff
@@ -0,0 +1,435 @@
+#include "Game.h"
+#include <fstream>
+#include <cstdlib>
+#include <iostream>
+
+int readHighScore() 
+{
+    int highScore = 0;
+    std::ifstream file(RECORD_FILE);
+    if (file.is_open()) {
+        file >> highScore;
+        file.close();
+    }
+    return highScore;
+}
+
+void writeHighScore(int highScore) 
+{
+    std::ofstream file(RECORD_FILE);
+    if (file.is_open()) {
+        file << highScore;
+        file.close();
+    }
+}
+
+Game::Game()
+    : window(sf::VideoMode(720, 920), "Doodle Jump", sf::Style::Default, sf::ContextSettings{ 8 }),
+    state(GameState::Menu),
+    menuState(window),
+    pauseState(window),
+    gameOverState(window),
+    minHeightGap(window.getSize().y * 0.08f),
+    maxHeightGap(window.getSize().y * 0.1f),
+    dy(0),
+    gravity(0.25f),
+    jumpVelocity(-12),
+    isJumping(false),
+    maxHeight(window.getSize().y * 0.4f),
+    hasBounced(false),
+    score(0),
+    movingPlatformSpawnProbability(0.1f),
+    highScore(readHighScore()) 
+{
+    initialize();
+}
+
+void Game::initialize() 
+{
+    window.setFramerateLimit(60);
+
+    if (!playerTexture.loadFromFile("doodle.png") ||
+        !platformTexture.loadFromFile("platform.png") ||
+        !brokenPlatformTexture.loadFromFile("broken_platform.png") ||
+        !movingPlatformTexture.loadFromFile("moving_platform.png") ||
+        !backgroundTexture.loadFromFile("background.png") ||
+        !font.loadFromFile("arial.ttf") ||
+        !buttonClickBuffer.loadFromFile("button_click.wav") ||
+        !gameOverBuffer.loadFromFile("game_over.mp3") ||
+        !collisionBuffer.loadFromFile("collision.mp3") ||
+        !pauseBuffer.loadFromFile("pause.mp3") ||
+        !platformBreakBuffer.loadFromFile("break.wav")) {
+        std::cerr << "Error loading resources" << std::endl;
+        exit(-1);
```

Comment:
I don't think errors in some of the textures should just close the game completely.
Just Missing pause.mp3 sound is much better then no game at all)